### ISSUE TITLE
cider--handle-debug: void function nrepl-push-input-handler

### ISSUE THREAD

**Issue by abo-abo** [clojure-emacs/cider#1051]
I can't get the debugger to work. I've instrumented a function, and I'm getting this error:

> error in process filter: cider--handle-debug: Symbol's function definition is void: nrepl-push-input-handler

I've been fighting with my config to get rid of the old nREPL warning.
I got this far:

> ; CIDER 0.9.0snapshot (package: 20150330.1124) (Java 1.7.0_75, Clojure 1.6.0, nREPL 0.2.10)

But I'm still getting:

> WARNING: The following required nREPL ops are not supported: 
> inspect-start inspect-reset
> Please, install (or update) cider-nrepl 0.9.0-SNAPSHOT and restart CIDER

**Comment by expez** [comment]
@abo-abo you need the latest snapshot of the middleware.  Do `lein -U repl` to force lein to check for snapshots.

**Comment by cichli** [comment]
Did you update CIDER without restarting emacs? That might be what's causing the missing ops message - the list of required ops is defined in a `defvar` which will have not been re-evaluated, so the now-removed ops are still there.

**Comment by abo-abo** [comment]
OK, I've restarted Emacs and with `lein -U repl` I get:


> nREPL server started on port 60748 on host 127.0.0.1 - nrepl://127.0.0.1:60748
> REPL-y 0.3.5, nREPL 0.2.10
> Clojure 1.6.0
> OpenJDK 64-Bit Server VM 1.7.0_75-b13

From `cider-connect` I get:

> ; CIDER 0.9.0snapshot (package: 20150330.1124) (Java 1.7.0_75, Clojure 1.6.0, nREPL 0.2.10)
> WARNING: The following required nREPL ops are not supported: 
> inspect-start inspect-reset
> Please, install (or update) cider-nrepl 0.9.0-SNAPSHOT and restart CIDER

For the instrumented function, I get an endless stdin request or a timeout.

**Comment by cichli** [comment]
Hmmm, `cider-nrepl-required-ops` should definitely not contain `inspect-start` or `inspect-reset` anymore - maybe check for any stale byte-compiled files? What does `cider-nrepl-required-ops` contain for you?

**Comment by abo-abo** [comment]
> What does cider-nrepl-required-ops contain for you?

That variable isn't defined. But I've finally solved the issue of having 3 different ciders on my
system. Now there's only one, and I did `make clean && make build` for it. Now the debugger works. Thanks for the suggestion.

Just one more minor issue: I'm still getting `nREPL 0.2.6` instead of `nREPL 0.2.10` using
`cider-jack-in` without a project.clj.  Is there a fix for this?
